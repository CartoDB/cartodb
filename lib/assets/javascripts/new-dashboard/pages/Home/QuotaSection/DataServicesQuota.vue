<template>
  <QuotaContainer :title="$t(`QuotaSection.dataServices`)" :perMonth="true">
    <QuotaWidget :name="$t(`QuotaSection.geocoding`)" :usedQuota="geocodingUsed" :availableQuota="geocodingAvailable" :helpLink="geocodingHelpLink"></QuotaWidget>
    <QuotaWidget :name="$t(`QuotaSection.isolines`)" :usedQuota="isolinesUsed" :availableQuota="isolinesAvailable" :helpLink="isolinesHelpLink"></QuotaWidget>
    <QuotaWidget :name="$t(`QuotaSection.routing`)" :usedQuota="routingUsed" :availableQuota="routingAvailable" :helpLink="routingHelpLink"></QuotaWidget>
  </QuotaContainer>
</template>

<script>
import { mapState } from 'vuex';
import QuotaWidget from './QuotaWidget';
import QuotaContainer from './QuotaContainer';

export default {
  name: 'DataServicesQuota',
  components: {
    QuotaWidget,
    QuotaContainer
  },
  computed: {
    ...mapState({
      geocodingUsed: state => state.user.geocoding.monthly_use,
      geocodingAvailable: state => state.user.geocoding.quota,
      routingUsed: state => state.user.mapzen_routing.monthly_use,
      routingAvailable: state => state.user.mapzen_routing.quota ? state.user.mapzen_routing.quota : 0,
      isolinesUsed: state => state.user.here_isolines.monthly_use,
      isolinesAvailable: state => state.user.here_isolines.quota
    }),
    geocodingHelpLink () {
      return 'https://carto.com/help/working-with-data/geocoding/';
    },
    isolinesHelpLink () {
      return 'https://carto.com/help/working-with-data/isolines/';
    },
    routingHelpLink () {
      return 'https://carto.com/help/working-with-data/routing/';
    }
  }
};
</script>
