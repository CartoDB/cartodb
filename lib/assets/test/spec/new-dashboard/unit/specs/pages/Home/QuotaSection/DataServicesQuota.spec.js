import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import usersArray from '../../../fixtures/users';
import DataServicesQuota from 'new-dashboard/pages/Home/QuotaSection/DataServicesQuota';

const localVue = createLocalVue();
localVue.use(Vuex);

let store = new Vuex.Store({
  state: {
    user: usersArray[2]
  }
});

const $tc = key => key;
const $t = key => key;

describe('DataServicesQuota.vue', () => {
  it('should render correct contents', () => {
    const dataServicesQuota = shallowMount(DataServicesQuota, {
      store,
      localVue,
      mocks: {
        $tc, $t
      }
    });

    expect(dataServicesQuota).toMatchSnapshot();
  });

  describe('Computed', () => {
    it('should return all the correct computed variables', () => {
      const dataServicesQuota = shallowMount(DataServicesQuota, {
        store,
        localVue,
        mocks: {
          $tc, $t
        }
      });

      const geocodingUsed = dataServicesQuota.vm.geocodingUsed;
      const geocodingAvailable = dataServicesQuota.vm.geocodingAvailable;
      const routingUsed = dataServicesQuota.vm.routingUsed;
      const routingAvailable = dataServicesQuota.vm.routingAvailable;
      const isolinesUsed = dataServicesQuota.vm.isolinesUsed;
      const isolinesAvailable = dataServicesQuota.vm.isolinesAvailable;

      expect(geocodingUsed).toBe(103);
      expect(geocodingAvailable).toBe(1200);
      expect(routingUsed).toBe(54);
      expect(routingAvailable).toBe(102);
      expect(isolinesUsed).toBe(134);
      expect(isolinesAvailable).toBe(1332);
    });
  });
});
