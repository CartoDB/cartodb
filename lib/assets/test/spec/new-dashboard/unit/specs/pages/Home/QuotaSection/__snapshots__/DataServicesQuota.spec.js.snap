// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataServicesQuota.vue should render correct contents 1`] = `
<quotacontainer-stub title="QuotaSection.dataServices" permonth="true">
  <quotawidget-stub name="QuotaSection.geocoding" availablequota="1200" usedquota="103" formattolocale="true" helplink="https://carto.com/help/working-with-data/geocoding/"></quotawidget-stub>
  <quotawidget-stub name="QuotaSection.isolines" availablequota="1332" usedquota="134" formattolocale="true" helplink="https://carto.com/help/working-with-data/isolines/"></quotawidget-stub>
  <quotawidget-stub name="QuotaSection.routing" availablequota="102" usedquota="54" formattolocale="true" helplink="https://carto.com/help/working-with-data/routing/"></quotawidget-stub>
</quotacontainer-stub>
`;
